### ISSUE TITLE
If you edit rancher compose, new upgrade_from value not taken into account

### ISSUE THREAD

**Issue by tfiduccia** [rancher/rancher#8787]
**Rancher versions:** master 5/17

**Steps to Reproduce:**
1.  Create a catalog template with 4 versions - 1.0.0 through 1.0.3
2.  Deploy template version 1.0.0
3.  Edit 1.0.2 template with upgrade_from = "1.0.1"
4.  Refresh catalog in Rancher so you can see the change
5.  Click on upgrade button for template

**Results:** I can still see 1.0.2 listed in the choose a version drop-down.

**Comments:** removing the upgrade_from also does not work.  But I figured this is the same thing.  But it should be checked before coming back to me.

**Comment by cjellick** [comment]
@tfiduccia in the bug report you say:
> Refresh catalog in Rancher so you can see the change
But in our offline convo, you clarified that you hit the "sync" button in rancher.
Can you confirm that here for the record? @amy thought you just hit refresh on the browser, which impacted her bug fix

**Comment by amy** [comment]
Update: Issue is that the UI never hits this endpoint `http://localhost:8080/v1-catalog/templates/test:apache:0` after I hit the refresh button on the Rancher UI and navigate back to the upgrade version link

**Comment by vincent99** [comment]
The templateVersion has already been loaded and are not reloaded because they are/were supposed to be immutable (but aren't really).  I'm sticking with don't do that™ for 1.6, this isn't a new problem.

**Comment by leodotcloud** [comment]
I have replaced `library` in a `v1.6.6` setup to point to my repo: `https://github.com/leodotcloud/rancher-catalog` and branch: `fix_dns_9547_9548`. I refreshed the catalog page and when I do a "Cmd+Click" on "Already Deployed" button of Network Services, I can see the new version that I just pushed out, i.e. v0.2.5. But I do not see this as an "Update Available" option in the Infrastructure Services page.

The "Upgrade Available" option started showing up after few minutes.
